Free/Type

```diff
@@ -0,0 +1 @@
+λ(f : Type → Type) → λ(a : Type) → ∀(r : Type) → (a → r) → (f r → r) → r
```

Comment:
This should now look like
```dhall
λ(f : Type → Type) → λ(a : Type) → ../../Mu/Type (../../FreeF/Type f a)
```

Also, it doesn’t exist on master yet, but I’ve started putting types like this (i.e., recursive types) into a “data/” directory, so this would then be “data/Free/Type”. There are some examples of this on the kind-polymorphism branch.